src/components/app/ProfileSharing.js

```diff
@@ -171,23 +187,40 @@ class ProfileSharingCompositeButton extends PureComponent<
     });
   }
 
-  _attemptToShare() {
-    if (this.state.state !== 'local' && this.state.state !== 'error') {
+  _attemptToShare(isSecondShare: boolean = false) {
+    if (
+      ((!isSecondShare && this.state.state !== 'local') ||
+        (isSecondShare && this.state.state !== 'public')) &&
+      this.state.state !== 'error'
+    ) {
       return;
     }
     this._notifyAnalytics();
 
-    const { profile, predictUrl } = this.props;
+    const { profile, predictUrl, profileSharingStatus } = this.props;
+    const { shareNetworkUrls } = this.state;
 
     new Promise(resolve => {
       if (!profile) {
         throw new Error('profile is null');
       }
-      const jsonString = serializeProfile(profile, this.state.shareNetworkUrls);
+      const urlShareOption = isSecondShare
+        ? !profileSharingStatus.sharedWithUrls
+        : shareNetworkUrls;
```

Comment:
Please add a comment explaining the rationale, that if it's the second share the checkbox is disabled (it wasn't very clear to me when reading the code as it's a bit hidden inside the `render` method) and so we take the value from profileSharingStatus. Also please explain why we don't use `sharedWithoutUrls`.

Reply:
Actually we don't need that `urlShareOption` option anymore since we have `_onSecondSharePanelOpen` that sets `this.state.shareNetworkUrls` to `
!this.props.profileSharingStatus.sharedWithUrls` when we open the secondary share panel. We can just pass the `this.state.shareNetworkUrls` to function.

I added `_onSecondSharePanelOpen` later and this slipped through my eyes.